### PR TITLE
Boost segmentation import performance

### PR DESCRIPTION
**Pull request by wonjuleee**
### Summary

When analyzing the import performance for `cityscapes` and `kaggle_image_mask`, I have checked that the most bottleneck is `np.unique` for parsing the unique class indices within each mask.
Analysis before PR:
image

Analysis after PR:
image

Instead of parsing unique class indices within a mask, I have changed to use all class indices in a dataset.
As a result, the performance is 5 times faster.

### Checklist

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [x] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by jihyeonyi** [comment]
Are you going to fix the unit-test errors? It seems that the errors are not directly related to this PR. Your code change looks good to me.

**Comment by wonjuleee** [comment]
> Are you going to fix the unit-test errors? It seems that the errors are not directly related to this PR. Your code change looks good to me.

The unit tests are broken because of checking `is_crowd` logic in https://github.com/openvinotoolkit/datumaro/blob/76fc9416e902ccd4e8bb364ed0d1cc2caef47785/src/datumaro/plugins/data_formats/cityscapes.py#L276-L283.

From my investigation, this is not defined in the official cityscapes document https://www.cityscapes-dataset.com/dataset-overview/#class-definitions, while this is for support CVAT cityscapes format as described in https://opencv.github.io/cvat/docs/manual/advanced/formats/format-cityscapes/. 
So, we need to decide whether drop this logic in Datumaro or not.
What do you think @openvinotoolkit/datumaro-maintainers?

File: src/datumaro/plugins/anchor_generator.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def get_anchors(
                 x_center + 0.5 * ws,
                 y_center + 0.5 * hs,
             ]
-            base_anchors = torch.stack(base_anchors, dim=-1)
+            base_anchors = torch.stack(base_anchors, dim=-1).to(self.device)
```
Comment (jihyeonyi) [inline src/datumaro/plugins/anchor_generator.py]: Is this related to the purpose of this PR?

**Comment by jihyeonyi** [comment]
> > Are you going to fix the unit-test errors? It seems that the errors are not directly related to this PR. Your code change looks good to me.
> 
> The unit tests are broken because of checking `is_crowd` logic in
> 
> https://github.com/openvinotoolkit/datumaro/blob/76fc9416e902ccd4e8bb364ed0d1cc2caef47785/src/datumaro/plugins/data_formats/cityscapes.py#L276-L283
> 
> .
> From my investigation, this is not defined in the official cityscapes document https://www.cityscapes-dataset.com/dataset-overview/#class-definitions, while this is for support CVAT cityscapes format as described in https://opencv.github.io/cvat/docs/manual/advanced/formats/format-cityscapes/. So, we need to decide whether drop this logic in Datumaro or not. What do you think @openvinotoolkit/datumaro-maintainers?

How about marking skip or xfail for this test and investigate this further later?
It is because we could miss something.

File: src/datumaro/plugins/anchor_generator.py (hunk 1 above)
Comment (wonjuleee) [reply src/datumaro/plugins/anchor_generator.py]: Nope, this is to fix anchor generator.. I will make a separate PR for fixing this. Thank you for pointing out this.

**Comment by wonjuleee** [comment]
> > > Are you going to fix the unit-test errors? It seems that the errors are not directly related to this PR. Your code change looks good to me.
> > 
> > 
> > The unit tests are broken because of checking `is_crowd` logic in
> > https://github.com/openvinotoolkit/datumaro/blob/76fc9416e902ccd4e8bb364ed0d1cc2caef47785/src/datumaro/plugins/data_formats/cityscapes.py#L276-L283
> > 
> > .
> > From my investigation, this is not defined in the official cityscapes document https://www.cityscapes-dataset.com/dataset-overview/#class-definitions, while this is for support CVAT cityscapes format as described in https://opencv.github.io/cvat/docs/manual/advanced/formats/format-cityscapes/. So, we need to decide whether drop this logic in Datumaro or not. What do you think @openvinotoolkit/datumaro-maintainers?
> 
> How about marking skip or xfail for this test and investigate this further later? It is because we could miss something.

I have fixed unit tests within cityscapes. Please review again :)

**Review by jihyeonyi — APPROVED** [review]
LGTM
